### PR TITLE
Modified modal config in firewall to prevent closing.

### LINKED ISSUE #2963: "New" Window Closes Without Prompt if User Clicks Outside the Window
If user clicks outside the "new" windows in various services, the windows gets closed and changes are lost. Can we implement a check if user (especially if this happens accidentally) clicks outside the box/window?

**Testflow:**
1. Login to Kura as admin/admin
2. Go to e.g. Firewall -> Port Forwarding -> New
3. Click on New button, enter several parameters
4. Without clicking "Submit", click somewhere outside the New window
5. Observe how window closes without a prompt if user wants to discard changes

**Expected behavior**
See above.

**Screenshot:**

**Target Environment:**
Board: Raspberry Pi 3
OS version: Linux raspberrypi 4.19.75-v7+ # 1270 SMP Tue Sep 24 18:45:11 BST 2019 armv7l GNU/Linux
Kura version: 4.2.0_SNAPSHOT ([Link to the specific Kura version -from June 29th, 2020](https://ci.eclipse.org/kura/job/kura-develop/lastSuccessfulBuild/artifact/kura/distrib/target/kura_4.2.0-SNAPSHOT_raspberry-pi-2-3_installer.deb)

### PR DESCRIPTION
**Pull request by MMaiero**
If the user clicks outside the modal.

Signed-off-by: Maiero <matteo.maiero@eurotech.com>

Closes #2963

### CODE REVIEW

**Comment by LeoNerdoG** [comment]
Hey @MMaiero,
I have checked this and is solves the issue in the Firewall service. 
May I suggest to fix this in all other service also? I think the best way is to fix all of these same issues in one PR. 
What do you think?

**Comment by MMaiero** [comment]
Do we have a list of the modals that should be tuned that way?

**Comment by LeoNerdoG** [comment]
I will make a list and post it here.

**Comment by LeoNerdoG** [comment]
Hey @MMaiero,
bellow you will find the list of modals. 

 - New button in Firewall -> Open Ports
 - Edit button in Firewall -> Open Ports
 - New button in Firewall -> Port Forwarding
 - Edit button in Firewall -> Port Forwarding
 - New button in Firewall -> IP Forwarding/Masquerading
 - Edit button in Firewall ->  IP Forwarding/Masquerading
 - New Connection button in Cloud Connections
 - New Pub/Sub button in Cloud Connection
 - New Driver button in Drivers and Assets
 - New Asset button in Drivers and Assets
 - Upload Channels button in Drivers and Assets -> Asset 
 - WireAsset asset in Wire graph
 - Install/Upgrade button in packages
 - Upload and Apply button in Settings -> Snapshot

**Comment by MMaiero** [comment]
Made few changes that should cover (hopefully) most of the modals reported to be updated

**Comment by LeoNerdoG** [comment]
OK I will check.

**Comment by LeoNerdoG** [comment]
hey @MMaiero, 
I have checked this PR and everything looks fine except the  "WireAsset asset in Wire graph" - this one can stille be closed with clikc outside the new window. Is this deliberate? Can you please check?

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/AlertDialog.ui.xml [hunk 1]
```diff
@@ -15,7 +15,8 @@
     xmlns:b="urn:import:org.gwtbootstrap3.client.ui" xmlns:b.html="urn:import:org.gwtbootstrap3.client.ui.html"
     xmlns:g="urn:import:com.google.gwt.user.client.ui" xmlns:gwt="urn:import:org.gwtbootstrap3.client.ui.gwt">
 
-    <b:Modal ui:field="alert" addStyleNames="confirm-dialog">
+    <b:Modal closable="false" fade="true" dataBackdrop="STATIC"
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/AlertDialog.ui.xml:18]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/ServicesUi.java [hunk 2]
```diff
@@ -41,6 +41,7 @@
 import org.gwtbootstrap3.client.ui.NavPills;
 import org.gwtbootstrap3.client.ui.PanelBody;
 import org.gwtbootstrap3.client.ui.TextBox;
+import org.gwtbootstrap3.client.ui.constants.ModalBackdrop;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/ServicesUi.java:44]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudConnectionsUi.java [hunk 3]
```diff
@@ -32,6 +32,7 @@
 import org.gwtbootstrap3.client.ui.ModalHeader;
 import org.gwtbootstrap3.client.ui.Panel;
 import org.gwtbootstrap3.client.ui.TabListItem;
+import org.gwtbootstrap3.client.ui.constants.ModalBackdrop;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudConnectionsUi.java:35]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudInstancesUi.java [hunk 4]
```diff
@@ -584,18 +580,7 @@ private void deleteConnection(final String factoryPid, final String cloudService
...
-        header.setTitle(MSGS.warning());
-        modal.add(header);
-
-        ModalBody body = new ModalBody();
-        body.add(new Span(MSGS.cloudServiceDeleteConfirmation()));
-        modal.add(body);
-
-        ModalFooter footer = new ModalFooter();
-        Button yes = new Button(MSGS.yesButton(), event -> {
+        alertDialog.show(MSGS.cloudServiceDeleteConfirmation(), () -> {
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudInstancesUi.java:588]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudInstancesUi.ui.xml [hunk 5]
```diff
@@ -67,7 +67,8 @@
                         width="100%" ui:field="connectionsGrid" />
                 </b:Well>
             </b:Row>
-            <b:Modal ui:field="newConnectionModal" b:id="newConnectionModal" title="{msgs.newCloudService}">
+            <b:Modal closable="false" fade="true" dataBackdrop="STATIC"
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/cloudconnection/CloudInstancesUi.ui.xml:70]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/AssetConfigurationUi.ui.xml [hunk 6]
```diff
@@ -70,7 +70,7 @@
                     </g:ScrollPanel>
                 </b:Well>
             </b:Row>
-            <b:Modal closable="true" fade="true" dataBackdrop="STATIC" dataKeyboard="true" b:id="new-channel-modal"
+            <b:Modal closable="false" fade="true" dataBackdrop="STATIC" dataKeyboard="true" b:id="new-channel-modal"
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/AssetConfigurationUi.ui.xml:72]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/DriversAndAssetsUi.ui.xml [hunk 7]
```diff
@@ -37,8 +37,8 @@
             </b:Row>
             <da:DriversAndAssetsListUi ui:field="driverAndAssetsListUi" />
 
-            <b:Modal closable="true" dataKeyboard="true" ui:field="newDriverModal"
-                title="{msgs.createNewDriverLabel}" b:id="newDriverModal">
+            <b:Modal closable="false" fade="true" dataBackdrop="STATIC"
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/drivers/assets/DriversAndAssetsUi.ui.xml:40]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkInterfacesTableUi.java [hunk 8]
```diff
@@ -25,6 +25,7 @@
 import org.gwtbootstrap3.client.ui.Modal;
 import org.gwtbootstrap3.client.ui.ModalBody;
 import org.gwtbootstrap3.client.ui.ModalFooter;
+import org.gwtbootstrap3.client.ui.constants.ModalBackdrop;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkInterfacesTableUi.java:28]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/CommandUserTabUi.java [hunk 9]
```diff
@@ -36,6 +36,7 @@
 import org.gwtbootstrap3.client.ui.ModalBody;
 import org.gwtbootstrap3.client.ui.ModalFooter;
 import org.gwtbootstrap3.client.ui.ModalHeader;
+import org.gwtbootstrap3.client.ui.constants.ModalBackdrop;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/CommandUserTabUi.java:40]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotsTabUi.ui.xml [hunk 10]
```diff
@@ -50,8 +50,8 @@
                     ui:field="snapshotsGrid" />
             </b:Row>
 			<!-- Modal -->
-            <b:Modal title="Upload" closable="true" fade="true"
-                b:id="uploadModal" ui:field="uploadModal">
+            <b:Modal closable="false" fade="true" dataBackdrop="STATIC"
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SnapshotsTabUi.ui.xml:53]: Please update header to "2020".

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SslTabUi.java [hunk 11]
```diff
@@ -36,6 +36,7 @@
 import org.gwtbootstrap3.client.ui.ModalBody;
 import org.gwtbootstrap3.client.ui.ModalFooter;
 import org.gwtbootstrap3.client.ui.ModalHeader;
+import org.gwtbootstrap3.client.ui.constants.ModalBackdrop;
```
Comment (LeoNerdoG) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/settings/SslTabUi.java:40]: Please update header to "2020".

**Review by LeoNerdoG — CHANGES_REQUESTED** [review]
hey @MMaiero can you please update some headers?

**Comment by MMaiero** [comment]
Done

**Review by LeoNerdoG — APPROVED** [review]

**Comment by LeoNerdoG** [comment]
Hey @MMaiero,
I ahve checked this PR and it looks OK. I think it can be merged in develop.

**Pull request merged**